lib/services/search_service.dart

```diff
@@ -132,26 +137,68 @@ class SearchService {
           CollectionsService.instance.getCollectionByID(file.collectionID);
       if (!collection.isArchived() &&
           collection.name.contains(nonCaseSensitiveRegexForQuery)) {
-        collectionSearchResults.add(CollectionWithThumbnail(collection, file));
+        collectionSearchResults
+            .add(AlbumSearchResult(CollectionWithThumbnail(collection, file)));
       }
     }
 
     return collectionSearchResults;
   }
 
-  Future<List<File>> getYearSearchResults(int year) async {
-    final yearInMicroseconds = DateTime.utc(year).microsecondsSinceEpoch;
-    final nextYearInMicroseconds =
+  Future<YearSearchResult> getYearSearchResults(int year) async {
+    final yearInMicrosecondsSinceEpoch =
+        DateTime.utc(year).microsecondsSinceEpoch;
+
+    final nextYearInMicrosecondsSinceEpoch =
         DateTime.utc(year + 1).microsecondsSinceEpoch;
-    final yearSearchResults =
-        await FilesDB.instance.getFilesCreatedWithinDurations(
+
+    final filesInYear = await FilesDB.instance.getFilesCreatedWithinDurations(
       [
-        [yearInMicroseconds, nextYearInMicroseconds]
+        [yearInMicrosecondsSinceEpoch, nextYearInMicrosecondsSinceEpoch]
       ],
       null,
       order: 'DESC',
     );
-    return yearSearchResults;
+    if (filesInYear.isEmpty) {
+      return null;
+    } else {
+      return YearSearchResult(year, filesInYear);
+    }
+  }
+
+  Future<List<HolidaySearchResult>> getHolidaySearchResults(
+    String query,
+  ) async {
+    final List<HolidaySearchResult> holidaySearchResult = [];
+
+    final nonCaseSensitiveRegexForQuery = RegExp(query, caseSensitive: false);
+
+    for (HolidayData holiday in allHolidays) {
```

Comment:
Dart can infer the type when the right-hand side is defined. For example, here you could simply use `for (final holiday in allHolidays)`